### PR TITLE
chore: upgrade project to node v20

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -299,7 +298,7 @@ class NodejsFunctionSanityAspect implements IAspect {
       if (!(node instanceof SolutionNodejsFunction)) {
         Annotations.of(node).addError('Directly using NodejsFunction is not allowed in the solution. Use SolutionNodejsFunction instead.');
       }
-      if (node.runtime != Runtime.NODEJS_18_X) {
+      if (node.runtime != Runtime.NODEJS_20_X) {
         Annotations.of(node).addError('You must use Nodejs 18.x runtime for Lambda with javascript in this solution.');
```
Comment (zxkane) [inline src/main.ts:302]: Update the error message also

**Review by zxkane — CHANGES_REQUESTED** [review]
- [x] update Node version in few github workflow

**Review by zxkane — CHANGES_REQUESTED** [review]
- [ ] update node version in project README and frontend's README

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -147,7 +147,7 @@ importers:
         version: 2.3.2(aws-cdk-lib@2.140.0)(constructs@10.3.0)
       cdk-nag:
         specifier: ^2.20.6
-        version: 2.28.109(aws-cdk-lib@2.140.0)(constructs@10.3.0)
+        version: 2.28.103(aws-cdk-lib@2.140.0)(constructs@10.3.0)
```
Comment (zxkane) [inline pnpm-lock.yaml:150]: revert those versions downgrade

**Review by zxkane — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (tyyzqmf) [reply pnpm-lock.yaml]: upgrade to 2.28.110

**Review by zxkane — APPROVED** [review]

**Pull request merged**